pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -171,6 +226,53 @@ func (am *Alertmanager) GetSilence(silence *types.Silence)    {}
 func (am *Alertmanager) CreateSilence(silence *types.Silence) {}
 func (am *Alertmanager) DeleteSilence(silence *types.Silence) {}
 
+func (am *Alertmanager) WorkingDirPath() string {
+	return filepath.Join(am.Settings.DataPath, workingDir)
+}
+
+// buildIntegrationsMap builds a map of name to the list of integration notifiers off of a list of receiver config.
+func (am *Alertmanager) buildIntegrationsMap(receivers []*config.Receiver, templates *template.Template) (map[string][]notify.Integration, error) {
+	integrationsMap := make(map[string][]notify.Integration, len(receivers))
+	for _, receiver := range receivers {
+		integrations, err := am.buildReceiverIntegrations(receiver, templates)
+		if err != nil {
+			return nil, err
+		}
+		integrationsMap[receiver.Name] = integrations
+	}
+
+	return integrationsMap, nil
+}
+
+// buildReceiverIntegrations builds a list of integration notifiers off of a receiver config.
+func (am *Alertmanager) buildReceiverIntegrations(receiver *config.Receiver, templates *template.Template) ([]notify.Integration, error) {
+	var (
+		errs         types.MultiError
+		integrations []notify.Integration
+		add          = func(name string, i int, rs notify.ResolvedSender, f func(l gokit_log.Logger) (notify.Notifier, error)) {
+			n, err := f(gokit_log.NewNopLogger())
+			if err != nil {
+				errs.Add(err)
+				return
+			}
+			integrations = append(integrations, notify.NewIntegration(n, rs, name, i))
+		}
+	)
+
+	for i, c := range receiver.WebhookConfigs {
+		add("webhook", i, c, func(l gokit_log.Logger) (notify.Notifier, error) { return webhook.New(c, templates, l) })
+	}
+
+	for i, c := range receiver.EmailConfigs {
+		add("email", i, c, func(l gokit_log.Logger) (notify.Notifier, error) { return email.New(c, templates, l), nil })
+	}
```

Comment:
Grafana will be using the new field as mentioned in another comment, so this would not exist

Reply:
Done!